lib/Dashboard/MailWidget.php

```diff
@@ -95,4 +124,88 @@ public function load(): void {
 			$this->accountService->findByUserId($this->userId)
 		);
 	}
+
+	/**
+	 * Get widget-specific search filter
+	 * @return string
+	 */
+	abstract public function getSearchFilter(): string;
+
+	/**
+	 * @param string $userId
+	 * @return Message[]
+	 * @throws ClientException
+	 * @throws ServiceException
+	 */
+	protected function getEmails(string $userId): array {
+		$accounts = $this->accountService->findByUserId($userId);
+		$filter = $this->getSearchFilter();
+		$emails = [];
+
+		foreach ($accounts as $account) {
+			$mailboxes = $this->mailManager->getMailboxes($account);
+			/** @var Mailbox[] $mailboxes */
+			$mailboxes = array_filter($mailboxes, function (Mailbox $mailbox) {
+				return $mailbox->isInbox();
+			});
+			foreach ($mailboxes as $mailbox) {
+				$searchResult = $this->mailSearch->findMessages(
```

Comment:
can't we use ``\OCA\Mail\Contracts\IMailSearch::findMessagesGlobally``? that eliminates looping over accounts and mailboxes

it sounds crazy but we have seen plenty of accounts where users have like 200 mailboxes. And if you loop over them this operation takes a lot of queries and time

Reply:
I didn't see this method. Much better indeed.